src/overlays/actors/ovl_En_Dodongo/z_en_dodongo.c

```diff
@@ -136,97 +280,839 @@ static DamageTable D_808792D4 = {
+            if (bomb != NULL) {
+                bomb->timer = 0;
+            }
+            this->unk_302 = 8;
+        }
+    } else if (func_801378B8(&this->skelAnime, 52.0f)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_GERUDOFT_DOWN);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_80877424.s")
+    if (this->unk_302 != 0) {
+        this->unk_302--;
+        if (this->unk_302 == 0) {
+            if (!this->actor.params) {
+                Item_DropCollectibleRandom(globalCtx, &this->actor, &this->actor.world.pos, 0x90);
+            } else {
+                Item_DropCollectible(globalCtx, &this->actor.world.pos, ITEM00_RUPEE_PURPLE);
+            }
+            Actor_MarkForDeath(&this->actor);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_80877494.s")
+void func_80878910(EnDodongo* this, GlobalContext* globalCtx) {
+    Vec3f sp3C;
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_80877500.s")
+    if (this->collider2.base.acFlags & AC_HIT) {
+        this->collider2.base.acFlags &= ~AC_HIT;
+        this->collider1.base.acFlags &= ~AC_HIT;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_808777A8.s")
+        for (i = 0; i < ARRAY_COUNT(this->collider2Elements); i++) {
+            if (this->collider2.elements[i].info.bumperFlags & BUMP_HIT) {
+                break;
+            }
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_8087784C.s")
+        if ((i != ARRAY_COUNT(this->collider2Elements)) &&
+            ((this->unk_300 != 10) || !(this->collider2.elements[i].info.acHitInfo->toucher.dmgFlags & 0xDB0B3))) {
+            func_80876D28(this, globalCtx);
+            Math_Vec3s_ToVec3f(&sp3C, &this->collider2.elements[i].info.bumper.hitPos);
+            if (this->actor.colChkInfo.damageEffect == 0xF) {
+                CollisionCheck_BlueBlood(globalCtx, NULL, &sp3C);
+                EffectSsHitMark_SpawnFixedScale(globalCtx, 0, &sp3C);
+            } else if (this->actor.colChkInfo.damageEffect != 14) {
+                EffectSsHitMark_SpawnFixedScale(globalCtx, 3, &sp3C);
+                CollisionCheck_SpawnShieldParticlesMetalSound(globalCtx, &sp3C, &this->actor.projectedPos);
+            }
+        }
+    } else if (this->collider1.base.acFlags & AC_HIT) {
+        this->collider2.base.acFlags &= ~AC_HIT;
+        this->collider1.base.acFlags &= ~AC_HIT;
+        func_800BE2B8(&this->actor, &this->collider1.base);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_80877D50.s")
+        for (i = 0; i < ARRAY_COUNT(this->collider1Elements); i++) {
+            if (this->collider1.elements[i].info.bumperFlags & BUMP_HIT) {
+                break;
+            }
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_80877D90.s")
+        if ((i != ARRAY_COUNT(this->collider1Elements)) &&
```

Comment:
Without documentation I'm not sure it's clear that this is actually `ARRAY_COUNT`: it may be a coincidence. Similarly elsewhere.

Reply:
Seems sensible to me, it's checking that i didn't reach the end of the array, because it's just about to use it as an index. If the loop never breaks, then it would index out of bounds.

Reply:
Fair enough, just making sure it wasn't actually the last collider that has some other property.